contracts/duckies/Garden.sol

```diff
@@ -121,52 +119,6 @@ contract Garden is Initializable, AccessControlUpgradeable, UUPSUpgradeable {
 		}
 	}
 
-	// -------- Payouts --------
-
-	function getReferralPayouts() external view returns (uint16[REFERRAL_MAX_DEPTH] memory) {
-		return _getReferralPayouts();
-	}
-
-	function setBaseReferralPayouts(
-		uint8[REFERRAL_MAX_DEPTH] memory referralPayouts
-	) public onlyRole(DEFAULT_ADMIN_ROLE) {
-		_baseReferralPayouts = referralPayouts;
-	}
-
-	function _getReferralPayouts() internal view returns (uint16[REFERRAL_MAX_DEPTH] memory) {
-		uint16[REFERRAL_MAX_DEPTH] memory referralPayouts = _baseReferralPayouts;
-		uint8 halvingDivider = halving();
-
-		if (halvingDivider >= 2 ** _MAX_HALVING_STEP) {
-			// return constant literal to save gas. Array length depends on REFERRAL_MAX_DEPTH value.
-			return [uint16(5), 4, 3, 2, 1];
-		}
-
-		for (uint i = 0; i < referralPayouts.length; i++) {
-			referralPayouts[i] /= halvingDivider;
-		}
-		return referralPayouts;
-	}
-
-	// -------- Halving --------
-
-	function halving() public view returns (uint8) {
```

Comment:
I suggest we remove halving and clean the code so we can merge, before adding changes

Reply:
Yes, I agree. Then, I'll separate this PR into 2.